nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/io/LogEntryCSVWriterTest.java

```diff
@@ -0,0 +1,114 @@
+import org.nuxeo.runtime.test.runner.Features;
+
+/**
+ * @since 11.1
+ */
+@Features(AuditFeature.class)
+@Deploy("org.nuxeo.ecm.platform.audit.tests:test-pageprovider-contrib.xml")
+public class LogEntryCSVWriterTest extends AbstractCSVWriterTest.External<LogEntryCSVWriter, LogEntry> {
+
+    @Inject
+    private PageProviderService pps;
+
+    @Inject
+    private CoreSession session;
+
+    public LogEntryCSVWriterTest() {
+        super(LogEntryCSVWriter.class, LogEntry.class);
+    }
+
+    @Test
+    public void testDefaultProperties() throws IOException {
+        DocumentModel root = session.getDocument(new PathRef("/"));
+        PageProvider<?> pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", null, Long.valueOf(20), Long.valueOf(0),
+                new HashMap<>(), root);
+        @SuppressWarnings("unchecked")
+        List<LogEntry> entries = (List<LogEntry>) pp.getCurrentPage();
+        // No fetch param in the context, retrieve all default properties
+        RenderingContext renderingCtx = RenderingContext.CtxBuilder.get();
+        for (LogEntry entry : entries) {
+            CSVAssert csv = csvAssert(entry, renderingCtx);
+            csv.has("id");
+            csv.has("category");
+            csv.has("principalName");
+            csv.has("comment");
+            csv.has("docLifeCycle");
+            csv.has("docPath");
+            csv.has("docType");
+            csv.has("docUUID"); // NOSONAR
+            csv.has("eventId");
+            csv.has("repositoryId"); // NOSONAR
+            csv.has("eventDate");
+            csv.has("logDate");
+        }
+    }
+
+    @Test
+    public void testCustomProperties() throws IOException {
+        DocumentModel root = session.getDocument(new PathRef("/"));
+        PageProvider<?> pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", null, Long.valueOf(20), Long.valueOf(0),
+                new HashMap<>(), root);
+        @SuppressWarnings("unchecked")
+        List<LogEntry> entries = (List<LogEntry>) pp.getCurrentPage();
+        // Fetching only specific properties via context param
+        RenderingContext renderingCtx = //
+                RenderingContext.CtxBuilder.fetch(LogEntryCSVWriter.ENTITY_TYPE, "id", "docUUID", "repositoryId").get();
+        for (LogEntry entry : entries) {
+            CSVAssert csv = csvAssert(entry, renderingCtx);
+            csv.has("id");
+            csv.has("docUUID");
+            csv.has("repositoryId");
+            // Default properties that are not requested and shouldn't be there
+            List<String> unwantedProps = List.of("category", "principalName", "comment", "docLifeCycle", "docPath",
+                    "docType", "eventId", "eventDate", "logDate");
+            for (String property : unwantedProps) {
+                try {
+                    csv.has(property);
+                    fail();
+                } catch (AssertionError e) {
+                    // should catch
```

Comment:
Your `fail` will be catched, and so your test will never fail on this part.

Reply:
Doh! 
Since `has` and `fail` will both give `AssertionError`
Unless I missed a hasNot kind of method. I will now rely on the error message. Fail gives null and has starts with "no field "